### DISCUSSION TITLE
File Structure Suggestions

### DISCUSSION THREAD

**Discussion by dizzyjaguar** [pmndrs/jotai#1472] [Q&A]
I saw that there was a discussion `https://github.com/pmndrs/jotai/discussions/896`, that was talking about best practices and such. I'm currently building a medium sized project and wondering if there are any suggestions for file structures. Mainly just curious how other people have gone about structuring their Jotai files and Atoms. 😄

**Comment by dai-shi** [comment]
File structure and export control is very important for atoms, because that's how we localize atoms. (so is custom hooks.)

In one of my projects, we have atoms in features and also global atoms that could be used across features.
```
src/features/
src/features/featureA/
src/features/featureA/atoms/
src/features/featureA/components/
src/features/featureB/
src/features/featureB/atoms/
src/features/featureB/components/
src/atoms/
```

**Reply by dizzyjaguar** [reply]
Thanks for the reply! 
A few questions more:

1. Currently I have all of my Atoms in a global file `src/atoms`. By not localizing them, is this going to affect anything performance-wise?
2. The project I'm using Jotai in right now is set up with `src/components` and `src/pages` rather than by Features. Do you have any suggestions for structuring my Jotai files in this type of file stucture?
thanks 😄

**Reply by dai-shi** [reply]
1: No, it's the same performance-wise. By localizing them, you can control where they are used, just like normal functions, components and custom hooks.
2: I'd still suggest to place local atoms in `src/components/...` and `src/pages/...` for larger apps.

**Reply by dizzyjaguar** [reply]
Okay great! thanks @dai-shi !
